pkg/operator/ceph/cluster/cluster.go

```diff
@@ -595,6 +600,47 @@ func (c *cluster) reportTelemetry() {
 	telemetry.ReportKeyValue(c.context, c.ClusterInfo, telemetry.ExternalModeEnabledKey, strconv.FormatBool(c.Spec.External.Enable))
 }
 
+func (c *cluster) generateMonAssimilateConf(clientName string, settings map[string]string) error {
+	root := path.Join(c.context.ConfigDir, c.Namespace)
+	assimilateConfPath := path.Join(root, fmt.Sprintf("%s.txt", clientName))
+	if err := os.MkdirAll(root, 0744); err != nil {
+		return errors.Wrapf(err, "failed to create assimilateConfPath for %s.", clientName)
+	}
+
+	cfg := `
+[` + clientName + `]
+`
+	err := ioutil.WriteFile(assimilateConfPath, []byte(cfg), 0700)
+	if err != nil {
+		rook.TerminateFatal(errors.Wrapf(err, "failed to write config file"))
+	}
+	configFile := ini.Empty()
+	s, err := configFile.NewSection(clientName)
+	if err != nil {
+		return err
+	}
+
+	for key, val := range settings {
+		if _, err := s.NewKey(key, val); err != nil {
+			return errors.Wrapf(err, "failed to add key %s", key)
+		}
+	}
+	if err := configFile.SaveTo(assimilateConfPath); err != nil {
+		return errors.Wrapf(err, "failed to save config file %s", assimilateConfPath)
+	}
+
+	args := []string{"config", "assimilate-conf", "-i", assimilateConfPath, "-o", fmt.Sprintf("%s.out", assimilateConfPath)}
```

Comment:
If it's just a simple append, I'd vote for the `+`:
```suggestion
	args := []string{"config", "assimilate-conf", "-i", assimilateConfPath, "-o", assimilateConfPath+".out"}
```